compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -354,8 +354,9 @@ void OMR::Power::MemoryReference::addToOffset(TR::Node * node, int64_t amount, T
       {
       self()->consolidateRegisters(NULL, NULL, false, cg);
       }
-   intptr_t displacement = self()->getOffset() + amount;
-   if (displacement<LOWER_IMMED || displacement>UPPER_IMMED)
+   int64_t displacement = self()->getOffset() + amount;
+   if (((displacement<LOWER_IMMED || displacement>UPPER_IMMED) && !cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10)) ||
+         (displacement < -0x200000000 || displacement > 0x1ffffffff))
       {
       TR::Register  *newBase;
       intptr_t     upper, lower;
```

Comment:
The handling of displacement is incorrect in the sense that it could be a 64-bit value now.  The existing code assumes it being a 32-bit value.

Reply:
It looks to me like the rest of the code does have correct handling of 64-bit displacements on 64-bit machines, as it uses `loadActualConstant` in that case. If you think there's still a problem, could you be more specific about what existing code is making this assumption?

Reply:
mis-reading the code ... it looks fine functionally.